lib/src/cupertino/cupertino_popover_menu.dart

```diff
@@ -235,21 +268,42 @@ class RenderPopover extends RenderShiftedBox {
     markNeedsPaint();
   }
 
+  /// Returns the size which must be reserved to display the arrow.
+  ///
+  /// When [extendAndClipContentOverArrow] is `true`, the [child] can take the space that is reserved for the arrow.
+  /// During paint, the [child] is clipped using the background shape.
+  ///
+  /// When [allowHorizontalArrow] is `false`, the reserved width is always zero.
+  Size get _reservedSizeForArrow => Size(
+        (allowHorizontalArrow ? arrowLength * 2 : 0),
+        (extendAndClipContentOverArrow ? 0.0 : arrowLength * 2),
+      );
+
+  /// Returns the background shape offset from the top left corner.
```

Comment:
This offset will very likely cause confusion. We should be extensive in our description about what it represents, to avoid confusion.

```
/// Returns the offset from the top-left corner of this `RenderObject` to the top-left
/// corner of the toolbar background shape.
///
/// The toolbar background shape is separated, for example, by the size of the arrow,
/// which might appear on the left side or top side of the toolbar background shape.
///
/// The arrow is always permitted to appear above and below the popup, therefore
/// space is always set aside above and below the toolbar background shape.
/// However, left and right arrows are only permitted when [allowHorizontalArrow] is
/// `true`. When it's `false`, no space is allocated to the left and right sides of the
/// toolbar background shape, within this `RenderObject`.
```